apps/src/lib/client/utils.rs

```diff
@@ -359,9 +359,20 @@ const TENDERMINT_NODE_ID_LENGTH: usize = 20;
 
 /// Derive Tendermint node ID from public key
 fn id_from_pk(pk: &common::PublicKey) -> TendermintNodeId {
-    let digest = Sha256::digest(pk.try_to_vec().unwrap().as_slice());
     let mut bytes = [0u8; TENDERMINT_NODE_ID_LENGTH];
-    bytes.copy_from_slice(&digest[..TENDERMINT_NODE_ID_LENGTH]);
+
+    match pk {
+        common::PublicKey::Ed25519(_) => {
+            let _pk: ed25519::PublicKey = pk.try_to_pk().unwrap();
+            let digest = Sha256::digest(_pk.try_to_vec().unwrap().as_slice());
+            bytes.copy_from_slice(&digest[..TENDERMINT_NODE_ID_LENGTH]);
+        },
+        common::PublicKey::Secp256k1(_) => {
+            let _pk: secp256k1::PublicKey = pk.try_to_pk().unwrap();
+            let digest = Sha256::digest(_pk.try_to_vec().unwrap().as_slice());
+            bytes.copy_from_slice(&digest[..TENDERMINT_NODE_ID_LENGTH]);
+        },
+    }
```

Comment:
Is there a more succinct way to write this code for the conversion from common to one of its variants?

Reply:
Could the following work?:
```
    match pk {
        common::PublicKey::Ed25519(_pk) => {
            let digest = Sha256::digest(_pk.try_to_vec().unwrap().as_slice());
            bytes.copy_from_slice(&digest[..TENDERMINT_NODE_ID_LENGTH]);
        },
        common::PublicKey::Secp256k1(_pk) => {
            let digest = Sha256::digest(_pk.try_to_vec().unwrap().as_slice());
            bytes.copy_from_slice(&digest[..TENDERMINT_NODE_ID_LENGTH]);
        },
    }
```
I.e. Somehow use pattern matching to obtain the contents you need instead of using `try_` functions?

Reply:
And if the above works and the borrow checker allows, you could try:
```
    let raw_bytes = match pk {
        common::PublicKey::Ed25519(_pk) => _pk.try_to_vec(),
        common::PublicKey::Secp256k1(_pk) => _pk.try_to_vec(),
    }
    let digest = Sha256::digest(raw_bytes.unwrap().as_slice());
    bytes.copy_from_slice(&digest[..TENDERMINT_NODE_ID_LENGTH]);
```